pkg/controller/tidbcluster/pod_control.go

```diff
@@ -412,6 +414,57 @@ func (c *PodController) syncTiKVPod(ctx context.Context, pod *corev1.Pod, tc *v1
 	return reconcile.Result{}, nil
 }
 
+func (c *PodController) syncTiDBPod(ctx context.Context, pod *corev1.Pod, tc *v1alpha1.TidbCluster) (reconcile.Result, error) {
+	value, ok := needDeleteTiDBPod(pod)
+	if !ok {
+		// No need to delete tidb pod
+		return reconcile.Result{}, nil
+	}
+
+	ns := tc.GetNamespace()
+	tcName := tc.GetName()
+
+	if tc.Status.PD.Phase == v1alpha1.UpgradePhase || tc.Status.PD.Phase == v1alpha1.ScalePhase ||
+		tc.Status.TiKV.Phase == v1alpha1.UpgradePhase || tc.Status.TiKV.Phase == v1alpha1.ScalePhase ||
+		tc.Status.TiFlash.Phase == v1alpha1.UpgradePhase || tc.Status.TiFlash.Phase == v1alpha1.ScalePhase ||
+		tc.Status.Pump.Phase == v1alpha1.UpgradePhase || tc.Status.Pump.Phase == v1alpha1.ScalePhase ||
+		tc.TiDBScaling() {
```

Comment:
Seems mixed checking directly against `tc.Status` and use methods like `tc.TiDBScaling()`. Is it feasible to:
1. use methods like `tc.PDUpgrading()` or `tc.PDScaling()` for each component instead of using `tc.Status` directly?
2. aggregate the checking for all components into a single method such as `tc.Upgrading()`?

That helps to make it easier for readability or reuse.

Reply:
Also curious is it intended to skip `tc.TiDBUpgrading` check for `tidb` only but keep the upgrading check for other components? If it is intended, maybe we can elaborate on the reason in a comment to avoid confusion.

Reply:
@hanlins Thanks  for the comments. I think I missed the tc.TiDBUpgrading case, added here and the test case. Also I used the suggested functions instead read tc status.